pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/parsers/MetsXmlParser.scala

```diff
@@ -0,0 +1,50 @@
+package uk.ac.wellcome.platform.transformer.mets.parsers
+
+import java.io.InputStream
+
+import uk.ac.wellcome.platform.transformer.mets.transformer.Mets
+
+import scala.util.{Failure, Success, Try}
+import scala.xml.{Elem, XML}
+
+object MetsXmlParser {
+
+  def apply(inputStream: InputStream): Try[Mets] =
+    for {
+      is <- Try(XML.load(inputStream))
+      mets <- MetsXmlParser(is)
+    } yield (mets)
+
+  def apply(root: Elem): Try[Mets] = {
+    for {
+      id <- recordIdentifier(root)
+      maybeAccessCondition <- accessCondition(root)
+    } yield
+      Mets(
+        recordIdentifier = id,
+        accessCondition = maybeAccessCondition,
+      )
+  }
+
+  private def recordIdentifier(root: Elem): Try[String] = {
+    val identifierNodes =
+      (root \\ "dmdSec" \ "mdWrap" \\ "recordInfo" \ "recordIdentifier").toList
+    identifierNodes match {
+      case List(identifierNode) => Success(identifierNode.text)
+      case _ =>
+        Failure(new Exception("Could not parse recordIdentifier from METS XML"))
```

Comment:
When these errors are throw in `process` - they won't have any identifiable information with them will there?

Could we add some? I can't remember what we get from the adapter - it's the S3 path?

Reply:
How about 

> ERROR u.a.w.p.t.m.s.MetsTransformerWorkerService - There was an error processing MetsData(mets.xml,6): blah blah blah..